### DISCUSSION TITLE
Unable to run game with manually patched exe

### DISCUSSION THREAD

**Discussion by tylerwill3000** [tsunamods-codes/7th-Heaven#227] [Q&A] [answered]
Ever since upgrading to 7th heaven version 4.0.0, I'm getting an error message on startup that my game appears to be pirated, and 7th heaven will therefore refuse to start. I reviewed the code changes for 4.0.0 and it appears this is due to some new code that was added to compare the hash of the FF7 EXE file against known legit copies.

Now, I know what you're thinking - you must be a pirate! Where'd you get your FF7 from? This bug report seems suspicious! However, I have a legit explanation for why I believe this is happening.

Recently, a member of the modding community (Dig) released a bugfix for the dreaded double-input bug on PC. I tried installing the 7th heaven IRO for this bugfix, but doing so created all kinds of technical difficulties on my end, such as the game refusing to output video when the modwas enabled (very strange). After about 4 days of unsuccessful troubleshooting with Dig, he eventually provided instructions for me to manually patch the EXE file so I could get the bugfix working instead of trying to apply it through the 7th heaven mod. After following his instructions, the double input bug seemed to be fixed on the 3.* version of 7th heaven. However, an unfortunate side effect of this manual patching is that my EXE file is now being flagged as a pirated copy on 7th heaven 4.*, despite my changes being totally legitimate. I've confirmed this manual patching is what's causing the EXE to be flagged since the error goes away if I revert back to the vanilla EXE.

Do I have any recourse to get the game working with a modified EXE, other than building my own version of 7th heaven from source with these checks removed or forever staying on the 3.* version? While I understand the motivation for adding these checks, it puts people like me (which are admittedly a minority) in a difficult position where I can't play the game with manually applied patches to workaround weird mod issues like the one I described above. I understand I could continue the dialog with Dig to see if there is either an adjustment he can make on his end or an adjustment I can make on my end to get it working, but as I stated above we already spent about 4 days debugging the issue to no avail so I don't expect I'd have much success there.

Again, I understand I'm in a unique position. One might say I've fallen through the cracks. Maybe my use case is not widespread enough to consider adding a fully supported workaround for this directly into 7th heaven, and I'll just have to figure out how to build a modified version from source. But, I'd really love to have a supported way to workaround this as I believe my use case (while rare) is legitimate. If nothing else, I could send you the manual HEX edits for the double input fix so you could compute the hash of the FF7 exe that has those specific changes applied and then add it to your allow list. I know this would be asking you guys to really go out of your way just for this one specific use case, but it would help me out a lot. I'm willing to do just about anything to get the double input fix working because of how much it improves the UX.

**Comment by julianxhokaxhiu** [comment, marked as answer]
Thanks for opening up this topic @tylerwill3000, I do understand the position you are on but we won't allow modified EXEs as it would open a wide door to also piracy. However to fix your own singular problem, you can either create a mod for 7th yourself using Hext to do the patches done on the exe, or you can ask Dig to create that for you. In either case you'd get back your functionality on top a genuine installation.

Hope this helps. Cheers!

**Reply by tylerwill3000** [reply]
Can you recommend a resource I can use as a guide to creating an FF7 mod that will use Hext to modify the EXE file? I'm a software developer by trade so I'd be willing to get my hands a bit dirty to create something like this, provided there's some documentation explaining how to make mods.

**Reply by julianxhokaxhiu** [reply]
You can start from here: https://7thheaven.rocks/help/modhelp.html it's not the most up-to-date but shall give you what you need. Also feel free to join our Discord channels as you'll find a very active community willing to help.

**Reply by tylerwill3000** [reply]
Thanks for the link. I'll explore my options from here. I also joined the discord. What would be the best channel for me to use if I have questions? Would it be "ff7-mod-support"?
